adagrams/game.py

```diff
@@ -1,11 +1,163 @@
+import random
+
+LETTER_POOL = {
+    'A': 9, 
+    'B': 2, 
+    'C': 2, 
+    'D': 4, 
+    'E': 12, 
+    'F': 2, 
+    'G': 3, 
+    'H': 2, 
+    'I': 9, 
+    'J': 1, 
+    'K': 1, 
+    'L': 4, 
+    'M': 2, 
+    'N': 6, 
+    'O': 8, 
+    'P': 2, 
+    'Q': 1, 
+    'R': 6, 
+    'S': 4, 
+    'T': 6, 
+    'U': 4, 
+    'V': 2, 
+    'W': 2, 
+    'X': 1, 
+    'Y': 2, 
+    'Z': 1
+}
+
+# =========== WAVE ONE ===========
+# ================================
+
 def draw_letters():
-    pass
+    
+    letters = []
+    
+    count = 0
+    
+    new_letter_dict = LETTER_POOL.copy()
+
+    while len(letters) < 10:
+        random_letter = random.choice(list(new_letter_dict.keys()))
+        for letter, points in new_letter_dict.items():
+            if letter == random_letter and points > 1:
+                letters.append(random_letter)
+                points -= 1
+                new_letter_dict[letter] = points
+            else:
+                continue
+        count += 1 
+    
+    return letters
+
+# =========== WAVE TWO ===========
+# ================================
 
 def uses_available_letters(word, letter_bank):
-    pass
+
+    word = word.upper()
+
+    list_of_words = list(word)
+
+    frequency_in_word = {}
+
+    for letter in list_of_words:
+        if letter in frequency_in_word:
+            frequency_in_word[letter] += 1
+        else:
+            frequency_in_word[letter] = 1
+
+    frequency_in_bank = {}
+
+    for letter in letter_bank:
+        if letter in frequency_in_bank:
+            frequency_in_bank[letter] += 1
+        else:
+            frequency_in_bank[letter] = 1
+
+
+    common = frequency_in_word.items() & frequency_in_bank.items() 
```

Comment:
This is a pretty clever way to compare two dictionaries, but there's a bit of a logic error here. `.items()` returns a collection of tuples of each key-value pair. And `&` will return the intersection of the two collections.

However, consider what would happen if `word` was `"add"` and `letter_bank` was `["A", "D", "D", "D"]`. In this case `frequency_in_word`'s entry for `"D"` would be 2, and `frequency_in_bank`'s entry would be 3. Since `("D", 2) != ("D", 3)` since the second element of the tuples are different, `&` would not include that in `common` and thus y'all's implementation will return `False`.

My suggestion implementation building off y'all's current one would be to go through each key-value pair in `frequency_in_word` in a `for` loop and then just make sure that the count of that letter in `frequency_in_bank` is greater than or equal to.

But this is a minor enough issue that our tests didn't catch that I'm not going to move down from a Green due to this.